backend/src/bundles/notifications/notification.repository.ts

```diff
@@ -0,0 +1,62 @@
+import { NotificationEntity } from '~/bundles/notifications/notification.entity.js';
+import { type NotificationModel } from '~/bundles/notifications/notification.model.js';
+import { type Repository } from '~/common/types/types.js';
+
+import { type UpdateNotificationRequestDto } from './types/types.js';
+
+class NotificationRepository implements Repository {
+    private notificationModel: typeof NotificationModel;
+
+    public constructor(notificationModel: typeof NotificationModel) {
+        this.notificationModel = notificationModel;
+    }
+
+    public findById(): ReturnType<Repository['findById']> {
+        return Promise.resolve(null);
+    }
+
+    public async findAll(): Promise<NotificationEntity[]> {
+        const notifications = await this.notificationModel
+            .query()
+            .where('isRead', false)
+            .execute();
+
+        return notifications.map((it) => NotificationEntity.initialize(it));
+    }
+
+    public async create(
+        entity: NotificationEntity,
+    ): Promise<NotificationEntity> {
+        const { type, isRead, userId } = entity.toNewObject();
+
+        const item = await this.notificationModel
+            .query()
+            .insert({
+                type,
+                isRead,
+                userId,
+            })
+            .returning('*')
+            .execute();
+
+        return NotificationEntity.initialize(item);
+    }
+
+    public async update(
+        notificationId: string,
+        payload: UpdateNotificationRequestDto,
+    ): Promise<NotificationEntity | null> {
+        const updatedItem = await this.notificationModel
+            .query()
+            .patchAndFetchById(notificationId, payload)
+            .execute();
+
+        return updatedItem ? NotificationEntity.initialize(updatedItem) : null;
+    }
+
+    public delete(): ReturnType<Repository['delete']> {
+        return Promise.resolve(true);
+    }
```

Comment:
You should either delete this method or provide a valid logic

Reply:
Repository and Service type force us to implement those methods, even if they are not used.